hr_applicant/models/hr_recruitment.py

```diff
@@ -0,0 +1,710 @@
+        if self.applicant_id and not self.relative_type:
+            warning = {
+                "title": _("Warning!"),
+                "message": _("Please select Relative Type!"),
+            }
+            return {"gender": False, "warning": warning}
+
+    @api.model
+    def create(self, vals):
+        if self._context.get("active_model") == "hr.applicant" and self._context.get(
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantRelative, self).create(vals)
+
+
+class ApplicantEducation(models.Model):
+    _name = "applicant.education"
+    _description = "Applicant Education"
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _rec_name = "from_date"
+    _order = "from_date"
+
+    from_date = fields.Date(string="From Date")
+    to_date = fields.Date(string="To Date")
+    education_rank = fields.Char("Education Rank")
+    school_name = fields.Char(string="School Name", size=256)
+    grade = fields.Char("Education Field/Major")
+    field = fields.Char(string="Major/Field of Education", size=128)
+    illiterate = fields.Boolean("Illiterate")
+    active = fields.Boolean(string="Active", default=True)
+    applicant_id = fields.Many2one(
+        "hr.applicant", "Applicant Ref", ondelete="cascade")
+    edu_type = fields.Selection(
+        [("Local", "Local"), ("Abroad", "Abroad")],
+        string="School Location",
+        default="Local",
+    )
+    country_id = fields.Many2one("res.country", "Country")
+    state_id = fields.Many2one("res.country.state", "State")
+    province = fields.Char("Province")
+
+    @api.onchange("edu_type")
+    def onchange_edu_type(self):
+        for rec in self:
+            if rec.edu_type == "Local":
+                rec.country_id = False
+            else:
+                rec.province = False
+                rec.state_id = False
+
+    @api.onchange("illiterate")
+    def onchange_illiterate(self):
+        for rec in self:
+            rec.from_date = False
+            rec.to_date = False
+            rec.education_rank = ""
+            rec.school_name = ""
+            rec.grade = ""
+            rec.field = ""
+            rec.edu_type = ""
+            rec.country_id = False
+            rec.state_id = False
+            rec.province = ""
+
+    @api.model
+    def create(self, vals):
+        if self._context.get("active_model") == "hr.applicant" and self._context.get(
+            "active_id"
+        ):
+            vals.update({"applicant_id": self._context.get("active_id")})
+        return super(ApplicantEducation, self).create(vals)
+
+    @api.onchange("from_date", "to_date")
+    def onchange_date(self):
+        to_date = self.to_date and datetime.strftime(
+            self.to_date, DEFAULT_SERVER_DATE_FORMAT
+        )
+        if (
+            to_date
+            and datetime.strptime(to_date, DEFAULT_SERVER_DATE_FORMAT)
+            >= datetime.today()
+        ):
```

Comment:
For Datetime conversion, Please use fields. Datetime class method.
like from_string, to_string method